src/NLog/Targets/FileTarget.cs

```diff
@@ -630,6 +635,18 @@ protected override void Write(LogEventInfo logEvent)
 #endif
             byte[] bytes = this.GetBytesToWrite(logEvent);
 
+            // Clean up old archives if this is the first time a log record has been written to
+            // this log file and the archiving system is date/time based.
+            if (this.ArchiveNumbering == ArchiveNumberingMode.Date && this.ArchiveEvery != FileArchivePeriod.None)
+            {
+                if (this.previousFileName != fileName)
```

Comment:
Would it be better then if it stored in some sort of collection the previous log filenames instead of only the last log filename? Then the archiver cleaner (`DeleteOldDateArchive`) runs if the log file that is being written to is not in the collection.